outputs/quickwit.go

```diff
@@ -0,0 +1,182 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package outputs
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/falcosecurity/falcosidekick/types"
+)
+
+type QuickwitDynamicMapping struct {
+	Description string `json:"description"`
+	Fast        bool   `json:"fast"`
+	ExpendDots  bool   `json:"expand_dots"`
+	Indexed     bool   `json:"indexed"`
+	Record      string `json:"record"`
+	Stored      bool   `json:"stored"`
+	Tokenizer   string `json:"tokenizer"`
+}
+
+type QuickwitFieldMapping struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+	Fast bool   `json:"fast"`
+}
+
+type QuickwitSearchSettings struct {
+	DefaultSearchFields []string `json:"default_search_fields"`
+}
+
+type QuickwitDocMapping struct {
+	DynamicMapping QuickwitDynamicMapping `json:"dynamic_mapping"`
+	FieldMappings  []QuickwitFieldMapping `json:"field_mappings"`
+	Mode           string                 `json:"mode"`
+	StoreSource    bool                   `json:"store_source"`
+	TimestampField string                 `json:"timestamp_field"`
+}
+
+type QuickwitMappingPayload struct {
+	Id             string                 `json:"index_id"`
+	Version        string                 `json:"version"`
+	SearchSettings QuickwitSearchSettings `json:"search_settings"`
+	DocMapping     QuickwitDocMapping     `json:"doc_mapping"`
+}
+
+func (c *Client) checkQuickwitIndexAlreadyExists(args types.InitClientArgs) bool {
+	config := args.Config.Quickwit
+
+	endpointUrl := fmt.Sprintf("%s/%s/indexes/%s/describe", config.HostPort, config.ApiEndpoint, config.Index)
+	quickwitCheckClient, err := InitClient("QuickwitCheckAlreadyExists", endpointUrl, config.MutualTLS, config.CheckCert, args)
+	if err != nil {
+		return false
+	}
+
+	if nil != quickwitCheckClient.sendRequest("GET", "") {
+		return false
+	}
+
+	return true
+}
+
+func (c *Client) AutoCreateQuickwitIndex(args types.InitClientArgs) error {
+	config := args.Config.Quickwit
+
+	if c.checkQuickwitIndexAlreadyExists(args) {
+		return nil
+	}
+
+	endpointUrl := fmt.Sprintf("%s/%s/indexes", config.HostPort, config.ApiEndpoint)
+	quickwitInitClient, err := InitClient("QuickwitInit", endpointUrl, config.MutualTLS, config.CheckCert, args)
+	if err != nil {
+		return err
+	}
+
+	mapping := &QuickwitMappingPayload{
+		Id:      config.Index,
+		Version: config.IndexVersion,
+		DocMapping: QuickwitDocMapping{
+			Mode:           "dynamic",
+			StoreSource:    true,
+			TimestampField: "time",
+			DynamicMapping: QuickwitDynamicMapping{
+				Description: "Falco",
+				Fast:        true,
+				ExpendDots:  true,
+				Indexed:     true,
+				Stored:      true,
+				Record:      "basic",
+				Tokenizer:   "raw",
+			},
+			FieldMappings: []QuickwitFieldMapping{
```

Comment:
It's done for `hostname` and `tags` but for a nested object we need to know the name of the subfields to be able to define a mapping:

```json
 "field_mappings": {
       "name": "foo",
        "type": "object"
        "field_mappings": {
             "name": "bar"
             "type": "text"
             "tokenizer": "raw"
        }
}
```

So we need to have some constants keys in the map to define a mapping. I think it would be the same for defining an Elasticsearch mapping :)

Reply:
Btw I think it'd be interesting to propose this feature (autocreate index with the mapping) for creating the Elasticsearch index, I can do that in a new PR too after this one :)

Reply:
@idrissneumann you can use a `json` field for the `output_fields`.

Reply:
Done @fmassot

Reply:
> Btw I think it'd be interesting to propose this feature (autocreate index with the mapping) for creating the Elasticsearch index, I can do that in a new PR too after this one :)

Yes sure, we just have the same issue with the `output_fields`, I let you create the PR later.

Reply:
@idrissneumann I notice the logs containing json are multi lines, can you use https://pkg.go.dev/encoding/json#Compact to minify them and have them on the same line please

Reply:
@Issif I printed some object values with the `%v` format which is not supposed to render multi-lined json and I checked with docker stdout/stderr like that:

<img width="1038" alt="Screenshot 2024-01-24 at 14 03 24" src="https://github.com/falcosecurity/falcosidekick/assets/12884347/7d8864e8-1edf-4797-a824-605d0c4a1ead">

Or like that:

<img width="1016" alt="Screenshot 2024-01-24 at 14 07 30" src="https://github.com/falcosecurity/falcosidekick/assets/12884347/bbbe8f94-bf68-405d-b46a-1f83a2e4fa8a">

Seems not multi-line to me, isn't it? Or you were referring to something else otherwise?

Reply:
> Done @fmassot . Just so you know, I also tried to add it as a default search field but it doesn't seems supported for json field.
> <img alt="Screenshot 2024-01-24 at 11 04 49" width="676" src="https://private-user-images.githubusercontent.com/12884347/299274482-6b46baaa-c29a-47e6-98e0-16d372fcadd9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxMDMyNDcsIm5iZiI6MTcwNjEwMjk0NywicGF0aCI6Ii8xMjg4NDM0Ny8yOTkyNzQ0ODItNmI0NmJhYWEtYzI5YS00N2U2LTk4ZTAtMTZkMzcyZmNhZGQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI0VDEzMjkwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ2ZmU3NTdhOTQwMDE2ZWU1ZTBkZDI4MDI4NDk2ZjkzMWQwZGJkMzViNjc1ODU1ODBhMzY0YzYwMTljYjM2MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xJ6MNJUVs9qTXIZsvVeiwMqnP1Khkv_DOSRZdc99-1g">

I saw it in that screenshot

Reply:
@Issif Those logs are coming from the common http client used by all the outputs and we said earlier we wanted to avoid refactoring it in this particular PR ^^

If you want I can also open an issue just saying the common http client need to inline the backend body response when there is an http error.

Reply:
Thanks for the merge. I'll now open three issues (for the `InitClient` method, for the elasticsearch's mapping and for this one) and start some PR :p